### ISSUE TITLE
architecture: unify model summary/result abstraction and rendering

### ISSUE THREAD

**Issue by TheHiddenObserver** [TheHiddenObserver/statgpu#154]
## Context

statgpu currently has several independent summary/result patterns rather than one repository-wide abstraction.

Examples from the maintained codebase:

- `statgpu.linear_model._stats.RegressionResults` owns both regression inference state and its own `summary()` rendering;
- Panel has a local shared design: `PanelSummary` is a structured dataclass, while `BasePanelModel._panel_summary()` constructs it and concrete estimators pass model-specific fields;
- `GeneralizedLinearModel`, `CoxPH`, GAM, Quantile wrappers, LogisticRegression, legacy penalized models, diagnostics, and other estimators still implement their own `summary()` formatting directly;
- CV wrappers often delegate `summary()` to the fitted base estimator rather than consuming a common result/rendering contract.

This means `summary()` is currently a public convention, not a unified architectural layer.

## Problem

The current design mixes three separate concerns in different ways across modules:

1. **statistical result state** — coefficients, covariance, standard errors, test statistics, p-values, confidence intervals, likelihood/AIC/BIC, fit statistics, diagnostics, convergence/provenance metadata;
2. **summary data assembly** — deciding which fields/sections belong in the report for a given estimator;
3. **presentation/rendering** — text tables, printing, string return values, structured dictionaries, and any future notebook/dataframe/HTML representation.

Because these concerns are not consistently separated:

- summary behavior differs across estimator families;
- some summaries print, some return strings, some return structured objects;
- feature naming/statistic labels/CI formatting/provenance handling are duplicated;
- model-specific extensions can require rewriting large formatting blocks;
- adding a new output representation would require touching many estimators;
- summary tests tend to be estimator-specific snapshots rather than contract tests against one shared representation.

## Goal

Design and implement a shared summary/result architecture that separates **model result semantics** from **rendering**, while preserving estimator-specific statistical content.

The desired direction is closer to:

```text
Fitted estimator / inference result
          │
          ▼
Structured summary data model
          │
          ├── common sections
          │     ├── model identity / call metadata
          │     ├── coefficient table
          │     ├── fit statistics
          │     ├── inference / covariance metadata
          │     └── convergence / backend / solver provenance
          │
          └── model-specific sections
                ├── Panel effects / variance components
                ├── Cox tie/baseline/survival metadata
                ├── GLM family/link/deviance
                └── other estimator-specific fields
          │
          ▼
Renderer(s)
    text / dict / dataframe / future HTML
```

This issue should **not** start by introducing one large `BaseSummary` class and forcing every estimator into an inheritance tree. Prefer composition: shared structured sections/data plus renderer(s), with model-specific sections added explicitly.

## Required first step: repository inventory

Before changing public behavior, inventory every maintained `summary()` / result object and classify it by current contract:

- return type;
- whether it prints as a side effect;
- coefficient/inference fields consumed;
- fit-statistic fields consumed;
- model-specific metadata;
- formula/feature-name behavior;
- covariance/statistic label behavior;
- convergence/backend/solver provenance;
- CV delegation behavior;
- current tests and documentation.

At minimum include:

- LinearRegression / `RegressionResults`;
- generalized linear models and typed GLM wrappers;
- LogisticRegression;
- Quantile and robust regression wrappers;
- penalized estimators where public summary exists;
- CoxPH / CoxPHCV;
- Panel estimators / `PanelSummary`;
- GAM;
- feature-selection / diagnostics summary surfaces that are intentionally public.

The inventory should explicitly distinguish true model summaries from unrelated helper objects that merely have a method named `summary()`.

## Proposed architecture questions

Resolve these before broad migration:

### 1. Structured result boundary

Decide whether the canonical object is:

- one generic immutable `SummaryResult` containing typed sections;
- a small common base protocol plus model-specific structured result objects;
- or another composition-oriented design.

Avoid using a generic `dict[str, Any]` as the only contract if that would make schema drift unauditable.

### 2. Relationship to inference results

statgpu already has structured inference result classes under `statgpu.inference._results`.

Determine whether summary data should:

- consume those objects directly;
- wrap them in a presentation-neutral summary model;
- or move additional fit/provenance data into a neighboring structured result layer.

Do not duplicate coefficient inference state into two independently authoritative objects.

### 3. Rendering contract

Define a renderer boundary such that the same structured result can support, where appropriate:

- deterministic plain text;
- `to_dict()` / JSON-friendly representation;
- dataframe/table representation;
- future notebook/HTML rendering without estimator changes.

Rendering must not change statistical values.

### 4. Estimator contract

Decide what public `model.summary()` returns.

Potential compatibility-preserving direction:

- estimator assembles a structured summary result;
- `summary()` returns that object;
- printing/string representation comes from the result/renderer;
- legacy print-on-call behavior, if retained temporarily, is explicitly deprecated or compatibility-wrapped rather than silently changed.

### 5. Model-specific extensions

The common contract must allow model-specific sections without making unrelated estimators carry meaningless nullable fields.

Examples:

- Panel: effects, variance components, theta, panel fit statistics;
- Cox: ties, strata/start-stop metadata, likelihood/baseline/survival information where applicable;
- GLM: family/link, deviance/dispersion, weighted objective metadata;
- penalized models: penalty identity, selected alpha, inference conditioning metadata;
- CV wrappers: selected hyperparameters and final-refit provenance.

## Panel as a reference design, not the universal template

The current Panel pattern is useful evidence:

- `PanelSummary` is structured;
- `BasePanelModel._panel_summary()` centralizes common construction;
- concrete estimators supply model-specific fields.

However, `PanelSummary` currently combines data and text rendering and contains several Panel-specific optional fields. Treat it as a migration reference, not as the repository-wide base class.

## Compatibility requirements

- Preserve statistical values and inference semantics exactly during structural migration.
- Do not change coefficient/covariance/df/AIC/BIC definitions merely to unify presentation.
- Preserve feature-name and formula alignment.
- Preserve model-specific applicability/failure behavior.
- Preserve sklearn-style fitted-state and clone contracts.
- CV wrappers must continue to summarize the actual selected/final fitted estimator and must not invent candidate-level results.
- Backend/device provenance exposed by current summaries must remain truthful.
- No silent loss of currently displayed fields.

Where current return/print behavior differs across models, document the compatibility plan before changing it.

### Stage A — inventory and schema

- complete repository-wide summary/result inventory;
- define the structured summary schema/protocol and renderer boundary;
- add contract tests for common coefficient/inference sections;
- no broad public-output migration yet.

### Stage B — migrate the most structured families

Good initial candidates:

- Panel (`PanelSummary` + `_panel_summary()`);
- Linear/GLM families that already expose structured inference state.

Preserve existing text formatting initially through compatibility renderers.

### Stage C — remaining model families

Migrate Cox, Quantile/robust, GAM, Logistic, penalized/CV summaries, then remove duplicated formatting helpers where safe.

## Validation

At minimum:

- golden/semantic tests proving pre/post migration statistical values are unchanged;
- deterministic text rendering tests;
- structured-result schema tests;
- `to_dict()`/serialization tests where supported;
- feature-name/formula tests;
- inference statistic/distribution label tests (`t` vs `z`, etc.);
- model-specific section tests;
- not-fitted and rank-deficient/unavailable-inference behavior;
- CV delegation/final-refit behavior;
- EN/CN documentation showing one coherent summary contract.

No physical GPU rerun should be required for presentation-only refactors unless numerical/inference code or backend execution paths are changed.

## Non-goals

- no repository-wide inference rewrite as part of this issue;
- no change to statistical definitions solely for formatting consistency;
- no requirement that every public object with a `summary()` method participate in the model-summary framework;
- no giant inheritance hierarchy whose subclasses override most behavior;
- no mandatory HTML/notebook renderer in the first implementation;
- no removal of model-specific summary sections in the name of uniformity.

## Acceptance criteria

- [ ] Every maintained model-summary surface is inventoried and classified.
- [ ] A documented structured summary/result contract separates statistical data assembly from rendering.
- [ ] The design composes common and model-specific sections without a monolithic nullable result type or giant `BaseSummary` override tree.
- [ ] Existing inference result objects have one clear relationship to the new summary layer; coefficient inference has a single authoritative source.
- [ ] Plain-text rendering is deterministic and backward-compatible where promised.
- [ ] At least Panel plus one non-Panel estimator family are migrated end to end before calling the shared architecture stable.
- [ ] Public `summary()` return/print semantics are documented consistently.
- [ ] Model-specific fields, provenance, formula feature names, and applicability behavior are preserved.
- [ ] Tests demonstrate no numerical/statistical behavior change from the summary refactor.
- [ ] EN/CN documentation describes the shared summary architecture and per-model extensions.

## Relationship to current work

This is a repository-wide architecture follow-up discovered while documenting PR #151. It should remain separate from PR #151 and from Panel-specific issue #93 because the duplication and inconsistent contracts span multiple estimator families.
